src/assembly/collections.ts

```diff
@@ -22,18 +22,22 @@ abstract class CollectionResult {
 }
 export class CollectionMutationResult extends CollectionResult {
   operation: string;
-  key: string;
+  keys: string[];
 
   constructor(
     collection: string,
     status: CollectionStatus,
     error: string,
     operation: string,
-    key: string,
+    keys: string[],
   ) {
     super(collection, status, error);
     this.operation = operation;
-    this.key = key;
+    if (keys == null) {
+      this.keys = [];
+    } else {
+      this.keys = keys;
+    }
```

Comment:
```suggestion
    this.keys = keys;    
```

`keys` can't be null here.